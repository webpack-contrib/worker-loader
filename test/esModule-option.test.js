import {
  compile,
  getCompiler,
  getErrors,
  getModuleSource,
  getResultFromBrowser,
  getWarnings,
} from "./helpers";

describe('"esModule" option', () => {
  it("should work and generate ES module syntax by default", async () => {
    const compiler = getCompiler("./basic/entry.js");
    const stats = await compile(compiler);
    // const result = await getResultFromBrowser(stats);

    expect(getModuleSource("./basic/worker.js", stats)).toMatchSnapshot(
      "module"
    );
    // expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should work with "true" value', async () => {
    const compiler = getCompiler("./basic/entry.js", {
      esModule: true,
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource("./basic/worker.js", stats)).toMatchSnapshot(
      "module"
    );
    expect(result).toMatchSnapshot("result");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should work with "false" value', async () => {
    const compiler = getCompiler("./basic/entry.js", {
      esModule: false,
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource("./basic/worker.js", stats)).toMatchSnapshot(
      "module"
    );
    expect(result).toMatchSnapshot("result");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
