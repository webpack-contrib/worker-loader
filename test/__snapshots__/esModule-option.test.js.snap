// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"esModule" option should work and generate ES module syntax by default: errors 1`] = `Array []`;

exports[`"esModule" option should work and generate ES module syntax by default: module 1`] = `
"export default function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\");
}
"
`;

exports[`"esModule" option should work and generate ES module syntax by default: warnings 1`] = `Array []`;

exports[`"esModule" option should work with "false" value: errors 1`] = `Array []`;

exports[`"esModule" option should work with "false" value: module 1`] = `
"module.exports = function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\");
}
"
`;

exports[`"esModule" option should work with "false" value: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"esModule" option should work with "false" value: warnings 1`] = `Array []`;

exports[`"esModule" option should work with "true" value: errors 1`] = `Array []`;

exports[`"esModule" option should work with "true" value: module 1`] = `
"export default function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\");
}
"
`;

exports[`"esModule" option should work with "true" value: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"esModule" option should work with "true" value: warnings 1`] = `Array []`;
